docs/documentation/soc-and-feeds.md

```diff
@@ -5,196 +5,411 @@ slug: /soc-and-feeds
 
-import Tabs from '@theme/Tabs'
-import TabItem from '@theme/TabItem'
 
-Swarm provides the ability to store content in content-addressed chunks or Single Owner Chunks (SOC). With single owner chunks, a user can assign arbitrary data to an address and attest chunk integrity with their digital signature.
+## Requirements
 
-Feeds are a unique feature of Swarm. They constitute the primary use case for single owner chunks. Developers can use Feeds to version revisions of a mutable resource, indexing sequential updates to a topic, publish the parts to streams, or post consecutive messages in a communication channel. Feeds implement persisted pull-messaging and can also be interpreted as a pub-sub system.
+Interactions with SOC and feeds requires the following:
 
-Because Feeds are built on top of SOCs, their interfaces are somewhat similar and use common concepts.
+* A fully initialized Bee light node running with fully synced postage batch data. (Running at `http://localhost:1633` by default)
+* A valid postage batch ID.
+* An Ethereum-compatible private key to sign updates. Using your node or blockchain account wallet's private key is strongly discouraged. Using a dedicated key for SOC / feeds is recommended. 
 
 ## Single Owner Chunks
 
-Bee-js calculates a SOC address as the hash of an `identifier` and `owner`. The `identifier` is a 32 bytes long arbitrary data, usually expected as a hex string or a `Uint8Array`. The `owner` is an Ethereum address that consists of 20 bytes in a format of a hex string or  `Uint8Array`.
+Bee-js calculates a SOC Swarm reference hash as the keccak256 hash of the concatenation of the  `identifier` and `owner` Ethereum address. The `identifier` is a 32 byte long arbitrary value (by default a hex string or a `Uint8Array`). The `owner` is an Ethereum address that consists of 20 bytes in a format of a hex string or `Uint8Array`.
+
+:::info
+SOCs are powerful and flexible low-level feature which provide the foundation upon which higher level abstractions such as [GSOC](/docs/gsoc/) and [feeds](/docs/soc-and-feeds/#feeds) are built. For most common use cases developers are recommended to use these higher level abstractions rather than interacting directly with SOCs themselves.
+:::
 
 :::warning SOCs are immutable!
-You might be tempted to modify a SOC's content to "update" the chunk. Reuploading of SOC is forbidden in Swarm as it might create uncertain behavior. Bee node will reject the API call if it finds already existing SOC for the given address. Either use a different `identifier`, or you might be looking for Feeds as your use case.
+You might be tempted to modify a SOC's content to "update" the chunk. Reuploading of an SOC is forbidden in Swarm as it might create uncertain behavior. A Bee node will reject the API call if it finds an already existing SOC for the given address. Either use a different `identifier`, or you might be looking for feeds if you need to perform multiple updates to the same content.
```

Comment:
Minor notes:

Reuploading an SOC is not forbidden, maybe "not recommended" instead.

Use "undefined" instead of "uncertain", as "undefined behaviour" is a well-recognized term in programming.

Reply:
> A Bee node will reject the API call if it finds an already existing SOC for the given address.

It does?

Reply:
> A Bee node will reject the API call if it finds an already existing SOC for the given address.

Actually I don't know about this part, this part was copied from the version of the warning from the original SOC section so I assumed it was correct?

Reply:
If this was the case, GSOC wouldn't work at all